qiskit_ibm_runtime/base_primitive.py

```diff
@@ -163,6 +164,15 @@ def _run(self, pubs: Union[list[EstimatorPub], list[SamplerPub]]) -> RuntimeJobV
 
         logger.info("Submitting job using options %s", primitive_options)
 
+        if options_dict.get("environment", {}).get("callback", None):
+            issue_deprecation_msg(
+                msg="The 'callback' option is deprecated",
+                version="0.37.0",
+                remedy="This option will have no effect since interim "
+                "results streaming was removed in a previous release.",
+                stacklevel=3,
+            )
```

Comment:
My understanding is that even though this option doesn't do anything, we still need to properly deprecate before removing.